### PR TITLE
Upnp changes

### PR DESCRIPTION
**Pull request by drder66**
This is my contribution to fixing issue #1547 (DLNA not working)
Covers many issues, described in the commit log.
Tested across a variety of devices, none of which worked properly (or at all) before these changes
Apologies if I haven't got things right, my first Github submission

### CODE REVIEW

**Comment by lachlan-00** [comment]
upnp was i think widely broken before you've had a look at this, really great to see how much effort has gone into this!

**Comment by lachlan-00** [comment]
fyi all i'm doing is working through the scrutinizer issues (which is why i'm committing over the top.)
https://scrutinizer-ci.com/g/ampache/ampache/inspections

becuase of the age of the code i think it'll be good to keep some of the old code commented out as you've done it so rearranging and making things match up with the style i've been keeping.

File: lib/class/upnp_api.class.php [hunk 1]
```diff
@@ -991,13 +1191,317 @@ public static function _videoChilds($prmPath, $prmQuery, $start, $count)
...
+        for ($i=0; $i < $actualsize; $i++) {
+            $actualtokens[$i]=trim($actualtokens[$i]);
+            if ($actualtokens[$i] != "") {
+                $nospacetokens[$h++] = $actualtokens[$i];
+            }
+        }
+
+        // now put together tokens which are actually one token e.g. upper hutt
+        $onetoken    = "";
+        $h           = 0;
```
Comment (lachlan-00) [inline lib/class/upnp_api.class.php:1221]: can we expand this variable name out to something?
i'm not familiar with the hutt word outside star wars?

Comment (drder66) [reply lib/class/upnp_api.class.php:1249]: Yes, that's not a very helpful comment.

// now put together the tokens which are not going to be search qualifiers

Would express the intent better

File: composer.json [hunk 2]
```diff
@@ -41,6 +41,7 @@
         "blueimp/jquery-file-upload": "10.2.0",
         "carhartl/jquery-cookie": "1.*",
         "cboden/ratchet": "0.4.*",
+        "clue/multicast-react": "dev-master",
```
Comment (usox) [inline composer.json:44]: HI. Could you specify a version constraint instead of using the master in here? That would be great

Comment (lachlan-00) [reply composer.json:44]: it looks like you willed it into existence. they only just release 16 hours ago
https://github.com/clue/reactphp-multicast/releases/tag/v1.1.0

**Comment by lachlan-00** [comment]
@drder66 i'm up to getting this merged now, are you able to pull my updates and check it out?

Was there any specific program/test you used to try all this out? it'd be good to actually write stuff down in the wiki if you have any process you followed

**Comment by lachlan-00** [comment]
i was also going to move the ssdp.service into examples and write up a upnp.md file in the docs folder so anything you can write about it would be really helpful for others

**Comment by drder66** [comment]
Is there anything else you need me to do?

**Comment by lachlan-00** [comment]
I'll be home by the weekend to check it all out again. Should be fine, as long as I can browses the server it's ready

**Pull request merged**